ethcore/src/client/client.rs

```diff
@@ -595,6 +594,15 @@ impl Client {
 		}
 	}
 
+	/// Take a snapshot.
+	pub fn take_snapshot<W: snapshot_io::SnapshotWriter + Send>(&self, writer: W) -> Result<(), ::error::Error> {
+		let db = self.state_db.lock();
```

Comment:
No need to lock the database here. It's fine for a command line snapshot, but for background snapshotting this should operate on a boxed clone